go/protocol/keybase1/extras.go

```diff
@@ -1115,6 +1115,20 @@ func (u UserPlusKeys) FindKID(needle KID) *PublicKey {
 	return nil
 }
 
+func (u UserPlusKeysV2AllIncarnations) FindKID(kid KID) (*UserPlusKeysV2, *PublicKeyV2NaCl) {
```

Comment:
For the behavior of this function, I think the "return `Current`'s version if there is one" is the really important bit, so that's good. Given that, it might also be slightly more consistent to traverse the past incarnations from front to back? For those PGP keys that get reused by multiple incarnations or something?